pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -39,60 +39,83 @@ def __init__(self, trainer, resume_from_checkpoint: Optional[Union[str, Path]] =
         self.loaded_checkpoint = dict()
         # used to validate checkpointing logic
         self.has_trained = False
+        self._load_optimizer_states = True
```

Comment:
reviewers: this is temporary. I don't know otherwise how to break down the large PR in #7652

Reply:
Should add a `FIXME` then

Reply:
added the fixme